app/app/controllers/application_controller.rb

```
@@ -59,6 +73,24 @@ def demo_mode?
     ENV["DOMAIN_NAME"] == "verify-demo.navapbc.cloud"
   end
 
+  def detect_client_agency_from_domain
+    return nil unless request.host.present?
+
+    agency_config.client_agency_ids.each do |agency_id|
+      agency = agency_config[agency_id]
+      if [ agency.agency_demo_domain, agency.agency_production_domain ].compact.include?(request.host)
+        return agency_id
+      end
+    end
+
+    raise "Unknown domain #{request.host}"
```

I figured we should raise an error here since it's configuration related